web/src/app/common/schema/form-types.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The type of parameter form field.
+ */
+export enum ParameterInputType {
+  Text = 'text',
+  File = 'file',
+}
+
+/**
+ * Types of hint message shown at the bottom of parameter forms.
+ */
+export enum ParameterHintType {
+  None = 'none',
+  Error = 'error',
+  Warning = 'warning',
+  Info = 'info',
+}
+
+/**
+ * The base type of parameter form fields.
+ */
+export interface ParameterFormFieldBase {
+  /**
+   * Unique ID of this parameter field.
+   */
+  id: string;
+  /**
+   * Type of this parameter field.
+   */
+  type: ParameterInputType;
+  /**
+   * Label of the parameter.
+   */
+  label: string;
+  /**
+   * Description of this parameter.
+   */
+  description: string;
+  /**
+   * Type of hint message (error,warning or info)
+   */
+  hintType: ParameterHintType;
+  /**
+   * The hint message shown at the bottom of inputs.
+   */
+  hint: string;
+}
+
+/**
+ * Text type parameter specific data in ParameterFormField type.
+ */
+export interface TextParameterFormField extends ParameterFormFieldBase {
+  type: ParameterInputType.Text;
+  /**
+   * If this text form field is editable from user or not.
+   */
+  allowEdit: boolean;
```

Comment:
Not a strong opinion so it's totally fine to keep this name. With that being said, just sharing my thoughts that the popular naming for this type of property might be "readonly"
https://developer.mozilla.org/ja/docs/Web/HTML/Attributes/readonly